### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by smithsz**
Thanks for your hard work, please ensure all items are complete before opening.

- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](https://github.com/cloudant/nodejs-cloudant/blob/master/DCO1.1.txt)
- [x] You have added tests for any code changes
- [x] You have updated the [CHANGES.md](https://github.com/cloudant/nodejs-cloudant/blob/master/CHANGES.md)
- [x] You have completed the PR template below:

## What
Update `README.md` for multiple plugin support.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -220,87 +220,99 @@ Cloudant({url:"companycloudant.local", username:"somebody", password:"somebody's
...
 
-#### The 'promises' Plugins
+   This plugin will automatically swap your Cloudant credentials for a cookie. It will handle the authentication for you and ensure that the cookie is refreshed when required.
+2. `promises`
 
-When initialising the Cloudant library, you can opt to use the 'promises' plugin:
+   If you'd prefer to write code in the _Promises_ style then the `promises` plugin turns each request into a _Promise_.
+3. `retry429` (or `retry`)
+
+   On occasion Cloudant's multi-tenant offering may reply with an HTTP 429 response because you've exceed the number of API requests in a given amount of time. The "retry429" plugin will automatically retry your request with an exponential back-off.
```
Comment (tomblench) [inline README.md]: retry429 should be in backticks in this para

File: README.md [hunk 2]
```diff
@@ -220,87 +220,99 @@ Cloudant({url:"companycloudant.local", username:"somebody", password:"somebody's
...
+   The plugin will automatically retry request errors (e.g. connection reset errors) with an exponential back-off.
+
+_Want to use multiple plugins?_ Not a problem. Simply pass the plugins as an array instead:
 
 ```js
-var cloudant = Cloudant({url: myurl, plugin:'promises'});
+var cloudant = new Cloudant({ url: myurl, plugins: [ 'cookieauth', 'promises', 'retry5xx' ] });
 var mydb = cloudant.db.use('mydb');
+mydb.get('mydoc', function(err, data) {
+  console.log(data.toString('utf8');
```
Comment (tomblench) [inline README.md]: Maybe something like `console.log("Document contents: "+data.toString('utf8');`

File: README.md [hunk 3]
```diff
@@ -220,87 +220,99 @@ Cloudant({url:"companycloudant.local", username:"somebody", password:"somebody's
...
+});
 ```
 
-Then the library will return a Promise for every asynchronous call:
+The plugins are _always_ executed in the order they are specified. Remember that all plugins are respected. If one requests a retry then it cannot be overruled by another. If two plugins request different delay times before the next retry attempt then the largest delay time is honoured.
+
+Be aware that if you don't specify any plugins then the `cookieauth` plugin will automatically be added. To disable all plugins you can pass an empty array as the plugin list, i.e. `Cloudant({ url: myurl, plugins: [] })`.
```
Comment (tomblench) [inline README.md]: Is the cookie auth plugin added if no username/password is set?

File: README.md [hunk 4]
```diff
@@ -220,87 +220,99 @@ Cloudant({url:"companycloudant.local", username:"somebody", password:"somebody's
...
+#### The `cookieauth` plugin
+
+For example:
 
 ```js
-mydb.list().then(function(data) {
-  console.log(data);
-}).catch(function(err) {
-  console.log('something went wrong', err);
+var cloudant = new Cloudant({ url: myurl, plugin: 'cookieauth' });
```
Comment (tomblench) [inline README.md]: Do we need to clarify that the user/pass is in the url?

File: README.md (hunk 3 above)
Comment (smithsz) [reply README.md]: y - it can still be added but doesn't do anything. will update docs accordingly.

File: README.md (hunk 4 above)
Comment (smithsz) [reply README.md]: good thinking. will use 'http://user:pass@localhost:5984' instead of `myurl` perhaps?

Comment (tomblench) [reply README.md]: sounds good

**Pull request merged**